### ISSUE TITLE
1.19 Support

### ISSUE THREAD

**Issue by Paradox-77** [DarkKronicle/KronHUD#63]
This mod is absolutely amazing, and have made this my main hud mod ever since its infancy days. I would love to continue using it in the newer versions of the game. Please do update the mod when you are able to. Thank you so much!

**Comment by moehreag** [comment]
This is currently not possible due to malilib not being available for 1.19.

**Comment by Paradox-77** [comment]
Oh yeah right, masa wants to rewrite all his mods. Forgot KronHUD needed malilib.

**Comment by TheKodeToad** [comment]
> Oh yeah right, masa wants to rewrite all his mods. Forgot KronHUD needed malilib.

sounds of pain

I hope I'm not expected to port it too DarkKore too.

**Comment by RoonMoonlight** [comment]
According to MaLiLib CurseForge page, MaLiLib code rewrite is ongoing

I guess this mod won't be updated unless they complete code rewrite process for MaLiLib

**Comment by Paradox-77** [comment]
There are already community ports of malilib for 1.19.
They can be found [here](https://kosma.pl/masamods/)

Any chance KronHUD might get a small patch for 1.19 with the community port of malilib?

**Comment by RoonMoonlight** [comment]
> There are already community ports of malilib for 1.19.
> They can be found [here](https://kosma.pl/masamods/)
> 
> Any chance KronHUD might get a small patch for 1.19 with the community port of malilib?

I don't think this is going to be updated unless MaLiLib officially gets 1.19 port

Community ports are unofficial and it should be considered risky

**Comment by Paradox-77** [comment]
> > There are already community ports of malilib for 1.19.
> > They can be found [here](https://kosma.pl/masamods/)
> > Any chance KronHUD might get a small patch for 1.19 with the community port of malilib?
> 
> I don't think this is going to be updated unless MaLiLib officially gets 1.19 port
> 
> Community ports are unofficial and it should be considered risky

Yes, I know the risk of using a community port, the ports are from a trusted and long-time member of the technical community. But as you said its not always a good idea trusting random strangers on the internet

**Comment by realbymaxgames** [comment]
yeah but please give us kronhud 1.19 with the community ported mailib.i don ´t wanna use a client for keystrokes

**Comment by allinkdev** [comment]
> > > There are already community ports of malilib for 1.19.
> > > They can be found [here](https://kosma.pl/masamods/)
> > > Any chance KronHUD might get a small patch for 1.19 with the community port of malilib?
> > 
> > 
> > I don't think this is going to be updated unless MaLiLib officially gets 1.19 port
> > Community ports are unofficial and it should be considered risky
> 
> Yes, I know the risk of using a community port, the ports are from a trusted and long-time member of the technical community. But as you said its not always a good idea trusting random strangers on the internet

I think the issue is more that the community port is unofficial, won't be updated/receive as much support as an in-house port by the developers.

Thus, I think we should just wait until it gets officially ported to 1.19 :grin: 

You can fork this mod, port everything non-MaLiLib to 1.19 since lots of mostly text-related stuff changed drastically and replace the MaLiLib version with the community port to 1.19 while we wait for it to be officially ported.

**Comment by TheKodeToad** [comment]
The mod was never actually "officially" updated to 1.19. I just opened a pull request and it was merged.

So if anyone does make a fork, I'm sure there's a possibility that it will be merged if a pull request is opened.

**Comment by lennareen** [comment]
masa said that he's abandoning the plan to have the "official" 1.19 mod versions that's based on the new code from 1.12.2, so the 1.19 malilib is still based on the old code. Although he'll publish testing versions of 1.19 based on the new code in his Discord server, it won't be uploaded to CurseForge.

**Comment by RoonMoonlight** [comment]
> masa said that he's abandoning the plan to have the "official" 1.19 mod versions that's based on the new code from 1.12.2, so the 1.19 malilib is still based on the old code. Although he'll publish testing versions of 1.19 based on the new code in his Discord server, it won't be uploaded to CurseForge.

That's great news! I thought it is better to seperate the development codebase (With new one and old one) while having compatibility with current MC version... Maybe he made the right choice

Now we can get update to support 1.19 in no time!

**Comment by TheKodeToad** [comment]
I think I will work on this unless moehreag wants too.

**Comment by moehreag** [comment]
> I think I will work on this unless moehreag wants too.
> 
> Unless DarkKronicle wants it to use DarkKore now.

Please do if you want, I'm not at home this week. (Unless the DarkKore case comes true)

**Comment by TheKodeToad** [comment]
I've done it.
Just need to wait for the pull request to be merged.

**Comment by Paradox-77** [comment]
Thank you so much @TheKodeToad, I had to use sodium extra's coord HUD until now. It's hideous compared to KronHUD

**Comment by DarkKronicle** [comment]
> > I think I will work on this unless moehreag wants too.
> > Unless DarkKronicle wants it to use DarkKore now.
> 
> Please do if you want, I'm not at home this week. (Unless the DarkKore case comes true)

DarkKore will probably be in the future if I get enough motivation
